test/models/role_test.rb

```diff
@@ -0,0 +1,20 @@
+require 'test_helper'
+
+class RoleTest < ActiveSupport::TestCase
+  test 'role requires a title' do
+    role = build(:role, title: '')
+
+    refute role.valid?
+  end
+
+  test 'title must be unique' do
+    standard_role = create :role, title: 'standard'
+    assert standard_role.valid?
+
+    duplicate_role = build :role, title: 'standard'
```

Comment:
Good question.  If you were to use `create` it would error silently, b/c validations are not met, and the test would fail.

By calling `build` then `save`, you are able to check for errors and validity, and the test continues to run its course.